core/src/main/java/org/infinispan/commands/remote/RevokeBiasCommand.java

```diff
@@ -0,0 +1,120 @@
+import org.infinispan.remoting.rpc.RpcManager;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.scattered.BiasManager;
+import org.infinispan.util.ByteString;
+
+/**
+ * Informs node that it is not allowed to serve reads from the local record anymore.
+ * After local bias is revoked a {@link BackupAckCommand} is sent to the originator.*
+ */
+//TODO: consolidate this with InvalidateVersionsCommand
+public class RevokeBiasCommand extends BaseRpcCommand {
+   public static final byte COMMAND_ID = 80;
+
+   private Address ackTarget;
+   private long id;
+   private int topologyId;
+   private Collection<Object> keys;
+   private Set<Integer> segments;
+   private transient BiasManager biasManager;
+   private transient CommandsFactory commandsFactory;
+   private transient RpcManager rpcManager;
+
+   public RevokeBiasCommand() {
+      super(null);
+   }
+
+   public RevokeBiasCommand(ByteString cacheName) {
+      super(cacheName);
+   }
+
+   public RevokeBiasCommand(ByteString cacheName, Address ackTarget, long id, int topologyId, Collection<Object> keys, Set<Integer> segments) {
+      super(cacheName);
+      this.ackTarget = ackTarget;
+      this.id = id;
+      this.topologyId = topologyId;
+      this.keys = keys;
+   }
+
+   public void init(BiasManager biasManager, CommandsFactory commandsFactory, RpcManager rpcManager) {
+      this.biasManager = biasManager;
+      this.commandsFactory = commandsFactory;
+      this.rpcManager = rpcManager;
+   }
+
+   @Override
+   public Object invoke() throws Throwable {
+      for (Object key : keys) {
+         biasManager.revokeLocalBias(key);
+      }
+      // ackTarget null means that this message is sent synchronously by primary owner == originator
+      if (ackTarget != null) {
+         BackupAckCommand backupAckCommand = commandsFactory.buildBackupAckCommand(id, topologyId);
+         rpcManager.sendTo(ackTarget, backupAckCommand, DeliverOrder.NONE);
+      }
+      return null;
+   }
+
+   @Override
+   public byte getCommandId() {
+      return COMMAND_ID;
+   }
+
+   @Override
+   public boolean isReturnValueExpected() {
+      return false;
+   }
+
+   @Override
+   public void writeTo(ObjectOutput output) throws IOException {
+      output.writeObject(ackTarget);
+      if (ackTarget != null) {
+         output.writeLong(id);
+      }
+      output.writeInt(topologyId);
+      if (keys != null) {
+         MarshallUtil.marshallCollection(keys, output);
+      } else {
+         assert segments != null;
+         MarshallUtil.marshallIntCollection(segments, output);
+      }
+   }
+
+   @Override
+   public void readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
+      ackTarget = (Address) input.readObject();
+      if (ackTarget != null) {
+         id = input.readLong();
+      }
+      topologyId = input.readInt();
+      keys = MarshallUtil.unmarshallCollection(input, ArrayList::new);
+      if (keys == null) {
```

Comment:
your luck is that the keys are never `null` otherwise you wouldn't be able to read the command.